### DISCUSSION TITLE
parameter passed to keyhandler turns up having an invalid address when being called

### DISCUSSION THREAD

**Discussion by kyomawolf** [codam-coding-college/MLX42#59] [Q&A] [answered]
So I'm not sure if it is a bug on your side, as I haven't worked on the rework of my project for a while, that's why I haven´t opened it as an issue.
[LINUX]
When the [keyhandler](https://github.com/mhahnFr/42cursus-miniRT/blob/71000bf504540b45c854201efdbc9f7ca63381dd/src/painter/key_handler.c#L54) function is being called the address of our pointer is set to 0x9, which results into a segvault.
![image](https://user-images.githubusercontent.com/69203043/200172661-c1d8d4d0-9405-454d-a7d8-03e46612830a.png)
[initialization](https://github.com/mhahnFr/42cursus-miniRT/blob/71000bf504540b45c854201efdbc9f7ca63381dd/src/main.c#L68)
![image](https://user-images.githubusercontent.com/69203043/200172767-56d84b15-a339-4094-9a50-bc6d57cc3ba2.png)
[related issue](https://github.com/mhahnFr/42cursus-miniRT/issues/1)
Did we miss something?

**Comment by W2Wizard** [comment, marked as answer]
That is the wrong prototype for the `key_handler` function.

See: https://github.com/codam-coding-college/MLX42/wiki/Hooks#key-hook
It needs to have a `struct` instead of an `int` for the first parameter.
